doc/api/errors.markdown

```diff
@@ -0,0 +1,448 @@
+# Errors
+
+<!--type=misc-->
+
+Errors generated by Node.js fall into two categories: JavaScript errors and system
```

Comment:
Changed *most* of these references. Retained "node style callback," since it seems disingenuous for io.js to claim credit for that pattern :)